llvm/include/llvm/CodeGen/SelectionDAGNodes.h

```diff
@@ -688,22 +687,6 @@ END_TWO_BYTE_PACK()
   /// \<target\>ISD namespace).
   bool isTargetOpcode() const { return NodeType >= ISD::BUILTIN_OP_END; }
 
-  /// Test if this node has a target-specific opcode that may raise
-  /// FP exceptions (in the \<target\>ISD namespace and greater than
-  /// FIRST_TARGET_STRICTFP_OPCODE).  Note that all target memory
-  /// opcode are currently automatically considered to possibly raise
-  /// FP exceptions as well.
-  bool isTargetStrictFPOpcode() const {
-    return NodeType >= ISD::FIRST_TARGET_STRICTFP_OPCODE;
-  }
-
-  /// Test if this node has a target-specific
-  /// memory-referencing opcode (in the \<target\>ISD namespace and
-  /// greater than FIRST_TARGET_MEMORY_OPCODE).
-  bool isTargetMemoryOpcode() const {
```

Comment:
Do we still need isTargetMemoryOpcode(). It doesn't have any callers.

Reply:
It has one use in the assert in `SelectionDAG::getMemIntrinsicNode`.

Reply:
Oops. I was looking in my local repo and should have checked for FIRST_TARGET_MEMORY_OPCODE to see that assert

Reply:
If nothing else is checking it, does that assert provide any value? Can we just allow any target opcode to be a memory opcode?

Reply:
I remembered it has one more "use". Before this patch it is implicitly checked [here](https://github.com/llvm/llvm-project/blob/73731d6873b6fb0757c3065aaf2452eaccd0eebc/llvm/lib/CodeGen/SelectionDAG/SelectionDAGISel.cpp#L4398). `isTargetStrictFPOpcode` returns true for memory opcodes as well. `mayRaiseFPException` is then used [here](https://github.com/llvm/llvm-project/blob/73731d6873b6fb0757c3065aaf2452eaccd0eebc/llvm/lib/CodeGen/SelectionDAG/SelectionDAGISel.cpp#L4227) to decide whether the resulting MachineInstr needs `NoFPExcept` flag. IIUC it will *not* set this flag on MachineInstrs if the matched SDNode(s) are flagged as memory accessing.

This behavior was introduced in 63336795. I couldn't conclude if it is safe to return false for memory opcodes so I left it as is.

After this patch the check is explicit, please see `SelectionDAGTargetInfo.cpp:mayRaiseFPException`.

Reply:
I ran CodeGen tests with the memory opcode check disabled and there was one failure in
`X86/fp-intrinsics-flags.ll`. The following diff resolves it:

![image](https://github.com/user-attachments/assets/c72af02c-a921-458a-8a9d-d05f3cf8e14c)

There was a discussion in https://reviews.llvm.org/D72466 that can be related to this. Can you confirm that this is a safe change? If so, then quoting @uweigand from https://reviews.llvm.org/D71841 I assume the check for memory opcode can be dropped:

> Note that there a bit of a quirk in identifying target nodes that are both memory nodes and strict FP nodes. To simplify the logic, right now all target memory nodes are automatically also considered strict FP nodes -- this could be fixed by adding one more range.